test/client/autofill.ts

```diff
@@ -0,0 +1,202 @@
+      }
+      this.mockRippled.addResponse('account_info', rippled.account_info.normal)
+      this.mockRippled.addResponse('ledger', rippled.ledger.normal)
+      this.mockRippled.addResponse('server_info', {
+        status: 'success',
+        type: 'response',
+        result: {
+          info: {
+            validated_ledger: {
+              base_fee_xrp: 0.00001,
+            },
+          },
+        },
+      })
+      const txResult = await this.client.autofill(tx)
+
+      assert.strictEqual(txResult.Fee, '399')
+    })
+
+    it('should autofill Fee of an AccountDelete transaction', async function () {
+      const tx: AccountDelete = {
+        Account: 'rf1BiGeXwwQoi8Z2ueFYTEXSwuJYfV2Jpn',
+        TransactionType: 'AccountDelete',
+        Destination: 'X7AcgcsBL6XDcUb289X4mJ8djcdyKaB5hJDWMArnXr61cqZ',
+      }
+      this.mockRippled.addResponse('account_info', rippled.account_info.normal)
+      this.mockRippled.addResponse('ledger', rippled.ledger.normal)
+      this.mockRippled.addResponse('server_info', {
+        status: 'success',
+        type: 'response',
+        result: {
+          info: {
+            validated_ledger: {
+              base_fee_xrp: 0.00001,
+            },
+          },
+        },
+      })
+      const txResult = await this.client.autofill(tx)
+
+      assert.strictEqual(txResult.Fee, '5000000')
+    })
+
+    it('should autofill Fee of an EscrowFinish transaction with signersCount', async function () {
+      const tx: EscrowFinish = {
+        Account: 'rf1BiGeXwwQoi8Z2ueFYTEXSwuJYfV2Jpn',
+        TransactionType: 'EscrowFinish',
+        Owner: 'rf1BiGeXwwQoi8Z2ueFYTEXSwuJYfV2Jpn',
+        OfferSequence: 7,
+        Condition:
+          'A0258020E3B0C44298FC1C149AFBF4C8996FB92427AE41E4649B934CA495991B7852B855810100',
+        Fulfillment: 'A0028000',
+      }
+      this.mockRippled.addResponse('account_info', rippled.account_info.normal)
+      this.mockRippled.addResponse('ledger', rippled.ledger.normal)
+      this.mockRippled.addResponse('server_info', {
+        status: 'success',
+        type: 'response',
+        result: {
+          info: {
+            validated_ledger: {
+              base_fee_xrp: 0.00001,
+            },
+          },
+        },
+      })
+      const txResult = await this.client.autofill(tx, 4)
+
+      assert.strictEqual(txResult.Fee, '459')
+    })
+  })
+
+  it("should autofill LastLedgerSequence when it's missing", async function () {
+    const tx: Transaction = {
+      TransactionType: 'DepositPreauth',
+      Account: 'rGWrZyQqhTp9Xu7G5Pkayo7bXjH4k4QYpf',
+      Authorize: 'rpZc4mVfWUif9CRoHRKKcmhu1nx2xktxBo',
+      Fee,
+      Sequence,
+    }
+    this.mockRippled.addResponse('ledger', {
+      status: 'success',
+      type: 'response',
+      result: {
+        ledger_index: 9038214,
+      },
+    })
+    const txResult = await this.client.autofill(tx)
+    assert.strictEqual(txResult.LastLedgerSequence, 9038234)
+  })
```

Comment:
I would add one more test that checks if everything works all together (if all of the autofillable fields are empty, will it autofill all of them correctly), given that you're using a lot of promises.